src/Microsoft.Health.Fhir.Core/Features/Context/FhirUserClaimParser.cs

```diff
@@ -36,8 +35,27 @@ public static class FhirUserClaimParser
                 }
             }
 
-            string fhirUserId = accessControlContext.FhirUserClaim.Segments.Last().TrimEnd('/');
-            string fhirUserResourceType = accessControlContext.FhirUserClaim.Segments[accessControlContext.FhirUserClaim.Segments.Length - 2].TrimEnd('/');
+            var fhirUserClaimUri = accessControlContext.FhirUserClaim;
+            if (!accessControlContext.FhirUserClaim.IsAbsoluteUri)
+            {
+                // convert to absolute Uri so that the Uri parsing methods will function
+                fhirUserClaimUri = new Uri(new Uri("http://fhirBase"), accessControlContext.FhirUserClaim);
+            }
+
+            if (fhirUserClaimUri.Segments.Length < 2)
```

Comment:
Can we also add another || condition to check if Segments length is greater than 2? fhirUserClaimUri.Segments.Length > 2?

Reply:
In the case of a fully qualified URL, the segments length may be longer than 2 and still be correct.